packages/core/src/test/codewhispererChat/tools/fsRead.test.ts

```diff
@@ -0,0 +1,118 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import assert from 'assert'
+import * as vscode from 'vscode'
+import { FsRead } from '../../../codewhispererChat/tools/fsRead'
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const TEST_DIR_NAME = 'fsread-tests'
+
+describe('FsRead Unit Tests', () => {
+    let rootUri: vscode.Uri
+    let testDirUri: vscode.Uri
+
+    before(async () => {
+        const workspaceFolders = vscode.workspace.workspaceFolders
+        if (!workspaceFolders || workspaceFolders.length === 0) {
+            throw new Error('No workspace folder is open. Please open a folder in VS Code before running tests.')
+        }
+        rootUri = workspaceFolders[0].uri
+
+        testDirUri = vscode.Uri.joinPath(rootUri, TEST_DIR_NAME)
+        await vscode.workspace.fs.createDirectory(testDirUri)
+    })
+
+    after(async () => {
+        await vscode.workspace.fs.delete(testDirUri, { recursive: true, useTrash: false })
+    })
+
+    it('reads entire file successfully (happy case)', async () => {
+        const fileUri = vscode.Uri.joinPath(testDirUri, 'fullFile.txt')
+        const fileContent = 'Line 1\nLine 2\nLine 3'
+        await vscode.workspace.fs.writeFile(fileUri, Buffer.from(fileContent, 'utf8'))
+
+        const fsRead = new FsRead({ path: fileUri.fsPath })
+        const result = await fsRead.invoke()
+
+        assert.strictEqual(result.output.kind, 'text', 'Output kind should be "text"')
+        assert.strictEqual(result.output.content, fileContent, 'File content should match exactly')
+    })
+
+    it('reads partial lines [2..4] of a file (happy case)', async () => {
```

Comment:
nitpick: you probably don't need to mention whether its a happy or sad test case in name

Reply:
Fixed.